### DISCUSSION TITLE
Treasury Economy API Support

### DISCUSSION THREAD

**Discussion by lokka30** [EssentialsX/Essentials#4808] [Ideas and Feature Suggestions] [answered]
Hi everyone,

[Treasury](https://www.spigotmc.org/resources/treasury.99531/) is a library that has an Economy API which EssentialsX can use.

Adopting Treasury will yield the following benefits to EssX:
* The API uses a CompletableFuture model - unlike when using Vault, plugins using an economy understand that retrieving info from the database is not instantaneous. This will mitigate any freezes on the main thread if the database is under load.
* Non-player accounts - plugins like Towny no longer need to use deprecated player methods to manage non-player balances.
* I assume EssX will remain single-currency, out of interest in stability and not complicating Vault support any further. But multi-currency support is still on the table, if that is desired.

Would it be ideal & feasible for this to be implemented? I'd like to hear what you all think of this possible integration.

Have a great day 🍻

**Comment by LcyDev** [comment]
I would like this

**Comment by Mineshafter61** [comment]
Supporting Treasury will unlock a whole plethora of new content possible with Essentials.

**Comment by lokka30** [comment]
Thanks for the community feedback so far. Treasury v2 was released some weeks ago which has refined things further.

**Comment by JRoy** [comment, marked as answer]
There is nothing stopping Treasury from using Vault and thus being supported by EssentialsX
